modules/dnn/include/opencv2/dnn/dnn.hpp

```diff
@@ -131,6 +148,47 @@ namespace dnn //! This namespace is used for dnn module functionlaity.
          */
         virtual int outputNameToIndex(String outputName);
 
+        /**
+         * @brief Ask layer if it support specific backend for doing computations.
+         * @param[in] backendId computation backend identifier.
+         * @see Backend
+         */
+        virtual bool supportBackend(int backendId);
+
+        /**
+         * @brief Returns pointers to Halide functions.
+         * @param[in] inputs Pointers to input Halide buffers.
+         *
+         * Input buffers should be exactly the same that will be used in forward invocations.
+         * Despite we can use Halide::ImageParam based on input shape only,
+         * it helps prevent some memory management issues (if something wrong,
+         * Halide tests will be failed).
+         */
+        virtual std::vector<void*> initHalide(const std::vector<void*> &inputs);
```

Comment:
Introduced `BackendNode` and `BackendWrapper` whose derivatives store explicit `Halide::Func` and `Halide::Buffer` correspondingly. Moreover, it helps remove `Halide::Func` members from layers as it was before.